tests/modules/test_securitydata.py

```diff
@@ -572,3 +610,311 @@ def test_saved_searches_returns_saved_search_client(
+        )
+        file_event_client = mocker.MagicMock(spec=FileEventClient)
+        file_event_client.search.return_value = file_event_search
+        file_event_client.get_file_location_detail_by_sha256.return_value = file_location
+        microservice_client_factory.get_file_event_client.return_value = file_event_client
+
+        pds_client = mocker.MagicMock(spec=PreservationDataServiceClient)
+        pds_client.find_file_versions.return_value = find_file_version
+        microservice_client_factory.get_preservation_data_service_client.return_value = pds_client
+
+        storage_node_client = mocker.MagicMock(spec=StoragePreservationDataClient)
+        storage_node_client.get_download_token.return_value = file_download
+        storage_node_client.get_file.side_effect = error_response
+        microservice_client_factory.create_storage_preservation_client.return_value = (
+            storage_node_client
+        )
+
+        with pytest.raises(Py42Error) as e:
+            security_module.stream_file_by_sha256("shahash")
+            assert e.value.args[0] == PDS_EXCEPTION_MESSAGE
+
+    def test_stream_file_by_md5_returns_stream_of_file(
+        self,
+        mocker,
+        security_client,
+        storage_client_factory,
+        microservice_client_factory,
+        file_event_search,
+        file_location,
+        find_file_version,
+        file_download,
+    ):
+
+        security_module = SecurityModule(
+            security_client, storage_client_factory, microservice_client_factory
+        )
+        file_event_client = mocker.MagicMock(spec=FileEventClient)
+        file_event_client.search.return_value = file_event_search
+        file_event_client.get_file_location_detail_by_sha256.return_value = file_location
+        microservice_client_factory.get_file_event_client.return_value = file_event_client
+
+        pds_client = mocker.MagicMock(spec=PreservationDataServiceClient)
+        pds_client.find_file_versions.return_value = find_file_version
+        microservice_client_factory.get_preservation_data_service_client.return_value = pds_client
+
+        storage_node_client = mocker.MagicMock(spec=StoragePreservationDataClient)
+        storage_node_client.get_download_token.return_value = file_download
+        storage_node_client.get_file.return_value = b"stream"
+        microservice_client_factory.create_storage_preservation_client.return_value = (
+            storage_node_client
+        )
+
+        response = security_module.stream_file_by_md5("md5hash")
+        assert response == b"stream"
+
+    def test_stream_file_by_md5_raises_file_not_found_error(
+        self,
+        mocker,
+        security_client,
+        storage_client_factory,
+        microservice_client_factory,
+        file_event_search,
+    ):
+        security_module = SecurityModule(
+            security_client, storage_client_factory, microservice_client_factory
+        )
+        file_event_search.text = "{}"
```

Comment:
Consider putting what the actual response looks like here for when the file isn't found/

Reply:
Or is it really an empty response?

Reply:
I guess, similar to a comment I made above, I don't understand the condition for this test

Reply:
This is a negative test, case when response content is empty.